File: walking_master/walking.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

int main()
{
    int case_num;
    scanf("%d", &case_num);
    for (int i = 0; i < case_num; i++)
    {
        int a, b, c, d;
        scanf("%d%d%d%d", &a, &b, &c, &d);
        int num_moves = 0;

        if (b > d)
        {
            printf("%d\n", -1);
            continue;
        }
        while (b != d)
        {
            a++;
            b++;
            num_moves++;
        }

        if (a < c)
        {
            printf("%d\n", -1);
            continue;
        }

        while (a != c)
        {
            a--;
            num_moves++;
        }
        printf("%d\n", num_moves);

    }
}

File: walking_master/walking.py
def main():
    case_num = int(input())
    for _ in range(case_num):
        a, b, c, d = input().split()
        num_moves = 0
        a = int(a)
        b = int(b)
        c = int(c)
        d = int(d)
        if b > d:
            print(-1)
            continue
        
        num_moves = d - b
        a = a + num_moves
        if a < c:
            print(-1)
            continue
        
        num_moves += (a - c)

        print(num_moves)
    
    
    
    
    
    
    
    
    
    
    
if __name__ == "__main__":
    main()